pageserver/src/tenant.rs

```diff
@@ -1690,8 +1693,8 @@ impl Tenant {
             .unwrap_or(self.conf.default_tenant_conf.trace_read_requests)
     }
 
-    pub fn update_tenant_config(&self, new_tenant_conf: TenantConfOpt) {
-        self.tenant_conf.write().unwrap().update(&new_tenant_conf);
+    pub fn set_new_tenant_config(&self, new_tenant_conf: TenantConfOpt) {
+        *self.tenant_conf.write().unwrap() = new_tenant_conf;
```

Comment:
Not quite sure what default values to strip off: `new_tenant_conf` comes from the `update_tenant_config_handler` function, where it's created with all `None` fields first: `let mut tenant_conf = TenantConfOpt::default();` and then filled with whatever comes from the API request.

Then we take this formed value and replace whatever was before in memory.
Good point that I have to do that only after the ondisk value is successfully updated, fixed that.

But otherwise, all new values that user had specified, even if they match the defaults, are still technically the overrides. Not specified user values are `None`, nothing to strip off.
Then we replace old values with the new values on disk and in memory, not leaving anything old, strippable behind?

Reply:
Aah yes oki makes sense. Oki, good point that now we can clear a field to use the default (because previously it would had been "don't change this").

Checking that "leave field out of PUT'd tenant_conf to use default" is covered in tests..

Reply:
That is the very end of the IT test I've changed, by adding extra HTTP assertions to similar values that were asserted before via pql.
Interestingly, that test mentions "trimming", but does not really check one..

Reply:
Yes, found it near the end, seemed allright a few hours ago.